apps/scan/frontend/src/screens/poll_worker_screen.tsx

```diff
@@ -154,88 +145,29 @@ export function PollWorkerScreen({
   const [pollWorkerFlowState, setPollWorkerFlowState] = useState<
     Optional<PollWorkerFlowState>
   >(initialPollWorkerFlowState());
-  const [currentPollsTransition, setCurrentPollsTransition] =
-    useState<PollsTransition>();
-  const [currentPollsTransitionTime, setCurrentPollsTransitionTime] =
-    useState<number>();
+
+  // optimistically set lastPollsTransition based on mutation input if it exists
```

Comment:
Why do we need the optimistic `lastPollsTransition` here?

Reply:
Because after the printing page, we land on the `transition_complete` page which needs the current transition to show accurate text. The other query will almost certainly be complete, but I don't want to bet on it.

Reply:
Maybe there's a way to encode that constraint more explicitly? E.g. block leaving the printing page until the mutation completes and the new transition info has loaded?

(Or at the very least a comment explaining this in a bit more detail would be helpful)

Reply:
I think that sort of block would require doing some sort of `await reactQuerySettledIfNotAlreadySetted(...)`, which is something I've wished for in the past but never found. And it's even a little more convoluted here because we're waiting on both a mutation and the subsequent query. I think you'd have to do some sort of `waitFor` which I wanted to avoid.

For now I'll just add a comment!

Reply:
Mutations can automatically wait for subsequent queries by awaiting the call to `queryClient.invalidateQueries` in the `onSuccess` handler, for future reference. So all you have to do is await the mutation to ensure that the invalidated queries are updated.

Reply:
But does that wait for those queries to be invalidated or wait until they've been invalidated and successfully refeched?

Reply:
I'm pretty sure it waits for them to be refetched: https://tkdodo.eu/blog/mastering-mutations-in-react-query#awaited-promises. But I could be wrong

Reply:
I just did some testing, and I don't think they do refetch in time. I removed the optimistic updating and the tests failed because we assert that the last polls transition exists before we render the transition screens, but it did not exist. If the refetching actually completed before `await mutateAsync` resolved, then it should just work as-is without the optimistic updating.

Reply:
comment: 64525cc64

Reply:
Dang, that's really surprising and unfortunate